File: src/algorithms/arrays/sliding-window/sliding-window-min-sum/sources/sliding-window-min-sum.rs
// Sliding Window Min Sum — O(n) minimum-sum subarray of fixed size
fn min_sum_subarray(input_array: &[i32], window_size: usize) -> (i32, usize) {
    if input_array.is_empty() || window_size == 0 || window_size > input_array.len() {
        // @step:initialize
        return (0, 0); // @step:initialize
    }

    // Compute the sum of the first window as the baseline
    let mut current_sum = 0i32; // @step:move-window
    for init_index in 0..window_size {
        // @step:move-window
        current_sum += input_array[init_index]; // @step:move-window
    }
    let mut min_sum = current_sum;
    let mut window_start_index = 0usize;

    // Slide the window: subtract left element, add right element
    for right_index in window_size..input_array.len() {
        current_sum -= input_array[right_index - window_size]; // @step:shrink-window
        current_sum += input_array[right_index]; // @step:expand-window

        if current_sum < min_sum {
            // @step:compare
            min_sum = current_sum; // @step:compare
            window_start_index = right_index - window_size + 1; // @step:compare
        }
    }

    (min_sum, window_start_index) // @step:complete
}

File: src/algorithms/arrays/sliding-window/sliding-window-min-sum/sources/sliding-window-min-sum.go
// Sliding Window Min Sum — O(n) minimum-sum subarray of fixed size
package slidingwindowminsum

func minSumSubarray(inputArray []int, windowSize int) (minSum int, windowStartIndex int) {
	if len(inputArray) == 0 || windowSize <= 0 || windowSize > len(inputArray) {
		// @step:initialize
		return 0, 0 // @step:initialize
	}

	// Compute the sum of the first window as the baseline
	currentSum := 0 // @step:move-window
	for initIndex := 0; initIndex < windowSize; initIndex++ { // @step:move-window
		currentSum += inputArray[initIndex] // @step:move-window
	}
	minSum = currentSum
	windowStartIndex = 0

	// Slide the window: subtract left element, add right element
	for rightIndex := windowSize; rightIndex < len(inputArray); rightIndex++ {
		currentSum -= inputArray[rightIndex-windowSize] // @step:shrink-window
		currentSum += inputArray[rightIndex]            // @step:expand-window

		if currentSum < minSum { // @step:compare
			minSum = currentSum                              // @step:compare
			windowStartIndex = rightIndex - windowSize + 1  // @step:compare
		}
	}

	return minSum, windowStartIndex // @step:complete
}
